examples/05-fetch-map/app.ts

```diff
@@ -43,13 +45,17 @@ const layerClasses: Record<LayerType, _ConstructorOf<Layer>> = {
 };
 function LayerFactory(layers: LayerDescriptor[]) {
   return layers
-    .map(({type, props}) => {
+    .map(({type, props, filters}) => {
       const LayerClass = layerClasses[type];
       if (!LayerClass) {
         console.error(`No layer class found for type: ${type}`);
         return null;
       }
-      return new LayerClass(props);
+      const filterProps = filters && {
+        ...getDataFilterExtensionProps(filters),
+        extensions: [new DataFilterExtension({filterSize: 4})],
+      };
+      return new LayerClass({...props, ...filterProps});
```

Comment:
I'm not sure if we have validation to ensure `filters` will be falsy if there are no filters applied, might need to do `import {isEmptyObject} from './utils.js';` and check that as well? Unless we think there are no side effects of an unused DataFilterExtension...